_posts/2017-08-01-0.1-using-network-policy.md

```diff
@@ -0,0 +1,139 @@
+
+Operating at the service layer gives the Envoy proxy a rich set of attributes to base policy decisions on, for protocols it understands, which at present includes HTTP/1.1 & HTTP/2. So, you can apply policy based on virtual host, URL, or other HTTP headers.  In the future, Istio will support a wide range of Layer 7 protocols, as well as generic TCP and UDP transport.
+
+In contrast, operating at the network layer has the advantage of being universal, since all network applications use IP. At the network layer you can apply policy regardless of the layer 7 protocol: DNS, SQL databases, real-time streaming, and a plethora of other services that do not use HTTP can be secured. Network Policy isn’t limited to a classic firewall’s tuple of IP addresses, proto, and ports. Both Istio and Network Policy are aware of rich Kubernetes labels to describe pod endpoints.
+
+## Implementation
+
+The Istio’s proxy is based on [Envoy](https://lyft.github.io/envoy/), which is implemented as a userspace daemon in the dataplane that interacts with the network layer using standard sockets. This gives it a large amount of flexibility in processing, and allows it to be distributed (and upgraded!) in a container.
+
+Network Policy dataplane is typically implemented in kernel space (e.g. using iptables, eBPF filters, or even custom kernel modules). Being in kernel space allows them to be extremely fast, but not as flexible as the Envoy proxy.
+
+## Enforcement Point
+
+Policy enforcement using the Envoy proxy is implemented inside the pod, as a sidecar container in the same network namespace. This allows a simple deployment model. However, since the proxy is in the same network namespace as the service instance, if the service is compromised, or misbehaves (as in a malicious tenant) the proxy can be bypassed. 
+
+While this won’t let an attacker access other Istio-enabled pods, so long as they are correctly configured, it opens several attack vectors:
+
+- Attacking unprotected pods
+- Attempting to deny service to protected pods by sending lots of traffic
+- Exfiltrating data collected in the pod
+- Attacking the cluster infrastructure (servers or Kubernetes services)
+- Attacking services outside the mesh, like databases, storage arrays, or legacy systems.
+
+Network Policy is typically enforced at the host node, outside the network namespace of the guest pods. This means that compromised or misbehaving pods cannot bypass policy enforcement. With the addition of egress policy due in Kubernetes 1.8, this difference makes Network Policy a key part of protecting your infrastructure from compromised workloads.
```

Comment:
You should probably elaborate on this point as the main advantage, the layered defense, since the first point about operating at different layers is not entirely true.